.github/workflows/test-js-packages.yml

```diff
@@ -203,6 +203,32 @@ jobs:
       - name: Run browser tests
         run: yarn workspace @noir-lang/noirc_abi test:browser
 
+  test-noir-js-backend-barretenberg:
+    needs: [build-noirc-abi]
+    name: noir-js-backend-barretenberg
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Download wasm package artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: noirc_abi_wasm
+          path: ./tooling/noirc_abi_wasm
+      
+      - name: Install Yarn dependencies
+        uses: ./.github/actions/setup
+
+      - name: Build noir_js_types
+        run: yarn workspace @noir-lang/types build
```

Comment:
(Not blocking): Why do we need this? Could we not build the types and then just download it like we do wasm-packages?

Not a blocker for this PR

Reply:
A couple reasons:
- It relies on the noirc_abi package artifact so we won't be parallelising anything if you're just considering the barretenberg wrapper tests.
- building this only takes 11 seconds so startup time for the runner becomes significant at that point.